src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/ODataServiceDocumentResourceV2Provider.cs

```diff
@@ -118,17 +123,17 @@ public ODataServiceDocumentResourceV2Provider()
                 throw new FatalProtocolException(message, ex);
             }
 
-            string serviceDocumentBaseAddress = null;
-            if (response.Status != HttpSourceResultStatus.NotFound && response.Status != HttpSourceResultStatus.NoContent)
+            // Trim any query string or trailing slash.
+            var baseAddress = lastRequestUri;
+            var firstQuestionMark = baseAddress.IndexOf('?');
+            if (firstQuestionMark >= 0)
             {
-                serviceDocumentBaseAddress = V2FeedParser.GetBaseAddress(response.Stream);
+                baseAddress = baseAddress.Substring(0, firstQuestionMark);
```

Comment:
There does not seem to be an API this (`AbsoluteUri` includes the query string). To avoid string methods (including concat), I will use `UriBuilder` to clear the query string.